src/ado/reproot.ado

```diff
@@ -138,52 +170,69 @@ qui {
         local tot_dirs = `tot_dirs' + `dirs'
       }
 
+      * Deduplicate the list in case the same root was found in multiple paths
+      local rootdirs : list uniq rootdirs
+
       * Output the grand total
       noi di as smcl `"{hline}"'
       noi di_search_results, total ///
-        time(`tot_time') dcount(`tot_dirs') rootdirs(`"`rootdirs'"')
+        time(`tot_time') dcount(`tot_dirs') rootdirs(`"`rootdirs'"') `verbose'
       noi di as smcl `"{hline}"'
 
 
       /***************************************************
         Parse the root files
       ***************************************************/
 
+      * List of roots found for this project
       local found_roots ""
 
+      * Parse all rootfile and filter out roots for this project
       foreach rootdir of local rootdirs {
+
+        * parse this root
         reproot_parse root, file("`rootdir'/`root_file'")
-        local this_root         "`r(root)'"
-        local this_root_global  "`prefix'`this_root'"
-        local this_root_project "`r(project)'"
+        local root = "`r(root)'"
 
-        * Test if this root belongs the relevant project
-        if "`project'" == "`this_root_project'" {
+        * Filter for this project and roots
+        if "`project'" == "`r(project)'" & `:list root in all_roots'  {
 
-          * Test if root was already found, if not then add to found_roots
-          if (`: list this_root in found_roots') {
-            noi di as error _n "{pstd}A second root called {result:`this_root)'} was found for this project found in folder {result:`rootdir'}.{p_end}"
+          *Test if root is duplicates
+          if (`:list root in found_roots') {
+            noi di as error `"{phang}Duplicate root found for root name [`root']. This root was found at [{it:`rootdir'}] after it had already been found at [{it:```prefix'`root'''}]. All roots must have a unique name within a project. No root globals were set.{p_end}"' _n
```

Comment:
```suggestion
            noi di as error `"{phang}Duplicate root found for root name [`root']. This root was found at [{browse `rootdir'}] after it had already been found at [{it:```prefix'`root'''}]. All roots must have a unique name within a project. No root globals were set.{p_end}"' _n
```

Reply:
I like and agree with this suggestion but implemented locally to have support for paths with `:` (all paths on Windows) and committed here: f9bcc0f406dc36971ea851d42bac7b24f9d6ab1a. Set as resolved if this looks good to you.